sw/device/silicon_creator/lib/cert/dice.c

```diff
@@ -86,9 +112,32 @@ static void curr_tbs_signature_le_to_be_convert(void) {
                    kAttestationSignatureBytes / 2);
 }
 
+/**
+ * Helper function to compute measurements of various OTP partitions that are to
+ * be included in attestation certificates.
+ */
+static status_t measure_otp_partition(otp_partition_t partition,
+                                      hmac_digest_t *measurement) {
+  otp_dai_read(partition, /*address=*/0, otp_state,
+               kOtpPartitions[partition].size / sizeof(uint32_t));
+  hmac_sha256(otp_state, kOtpPartitions[partition].size, measurement);
+  return OK_STATUS();
+}
+
 status_t dice_uds_cert_build(manuf_certgen_inputs_t *inputs,
                              hmac_digest_t *uds_pubkey_id, uint8_t *tbs_cert,
                              size_t *tbs_cert_size) {
+  // Measure OTP partitions.
+  // TODO(#19455): check against digests present in the OTP digest field.
+  hmac_digest_t otp_creator_sw_cfg_measurement = {.digest = {0}};
+  hmac_digest_t otp_owner_sw_cfg_measurement = {.digest = {0}};
+  hmac_digest_t otp_hw_cfg0_measurement = {.digest = {0}};
+  TRY(measure_otp_partition(kOtpPartitionCreatorSwCfg,
```

Comment:
Follow-up PR: we should not be using `TRY` in silicon_creator code.  We should be using either `RETURN_IF_ERROR` or `HARDENED_RETURN_IF_ERROR` from `silicon_creator/lib/error.h`.

Reply:
Same here.